jaxtronomy/LensModel/Solver/lens_equation_solver.py

```diff
@@ -0,0 +1,773 @@
+import copy
+
+import jax.lax as lax
+
+import jax.random as random
+import jax.numpy as jnp
+from jax import jit
+import numpy as np
+import jaxtronomy.Util.util as util
+import lenstronomy.Util.image_util as image_util
+from jax.scipy.optimize import minimize
+from lenstronomy.LensModel.Solver.epl_shear_solver import solve_lenseq_pemd
+
+__all__ = ["LensEquationSolver"]
+
+SUPPORTED_LENS_MODELS_ANALYTICAL = (
+    ["SIE", "SHEAR"],
+    ["SIE"],
+    ["EPL_NUMBA", "SHEAR"],
+    ["EPL_NUMBA"],
+    ["EPL", "SHEAR"],
+    ["EPL"],
+)
+
+
+class LensEquationSolver(object):
+    """Class to solve for image positions given lens model and source position."""
+
+    def __init__(self, lensModel):
+        """This class must contain the following definitions (with same syntax as the
+        standard LensModel() class: def ray_shooting() def hessian() def magnification()
+
+        :param lensModel: instance of a class according to
```

Comment:
or has this to be an instance of the jax version?

Reply:
Whoops, I was not careful enough about the documentation. This one is fixed now